src/cswinrt/strings/additions/Windows.Storage.Streams/WindowsRuntimeBufferExtensions.cs

```diff
@@ -59,58 +59,91 @@ public static IBuffer AsBuffer(this byte[] source, int offset, int length, int c
-
-            CopyTo(source, 0, destination, 0, source.Length);
-        }
-
-
-        /// <summary>
-        /// Copies <code>count</code> bytes from <code>source</code> starting at offset <code>sourceIndex</code>
-        /// to <code>destination</code> starting at <code>destinationIndex</code>.
-        /// This method does <em>NOT</em> update <code>destination.Length</code>.
-        /// </summary>
-        /// <param name="source">Array to copy data from.</param>
-        /// <param name="sourceIndex">Position in the array from where to start copying.</param>
-        /// <param name="destination">The buffer to copy to.</param>
-        /// <param name="destinationIndex">Position in the buffer to where to start copying.</param>
-        /// <param name="count">The number of bytes to copy.</param>
-        public static void CopyTo(this byte[] source, int sourceIndex, IBuffer destination, uint destinationIndex, int count)
-        {
-            if (source == null) throw new ArgumentNullException(nameof(source));
-            if (destination == null) throw new ArgumentNullException(nameof(destination));
-            if (count < 0) throw new ArgumentOutOfRangeException(nameof(count));
-            if (sourceIndex < 0) throw new ArgumentOutOfRangeException(nameof(sourceIndex));
-            if (source.Length < sourceIndex) throw new ArgumentException(global::Windows.Storage.Streams.SR.Argument_IndexOutOfArrayBounds, nameof(sourceIndex));
-            if (source.Length - sourceIndex < count) throw new ArgumentException(global::Windows.Storage.Streams.SR.Argument_InsufficientArrayElementsAfterOffset);
-            if (destination.Capacity < destinationIndex) throw new ArgumentException(global::Windows.Storage.Streams.SR.Argument_BufferIndexExceedsCapacity);
-            if (destination.Capacity - destinationIndex < count) throw new ArgumentException(global::Windows.Storage.Streams.SR.Argument_InsufficientSpaceInTargetBuffer);
-            if (count == 0) return;
-
-            // If destination is backed by a managed array, use the array instead of the pointer as it does not require pinning:
-            byte[] destDataArr;
-            int destDataOffs;
-            if (destination.TryGetUnderlyingData(out destDataArr, out destDataOffs))
-            {
-                global::System.Buffer.BlockCopy(source, sourceIndex, destDataArr, (int)(destDataOffs + destinationIndex), count);
-                return;
-            }
-
-            IntPtr destPtr = destination.GetPointerAtOffset(destinationIndex);
-            Marshal.Copy(source, sourceIndex, destPtr, count);
-        }
-
+#region (Span<Byte>).CopyTo extensions for copying to an (IBuffer)
+
+        /// <summary>
+        /// Copies the contents of <code>source</code> to <code>destination</code> starting at offset 0.
+        /// This method does <em>NOT</em> update <code>destination.Length</code>.
+        /// </summary>
+        /// <param name="source">Array to copy data from.</param>

```

Comment:
Should we update the parameter comments in the added methods to say `Span` instead of `Array`?